File: sleepy-test/CSharpXUnit/CSharpXUnit.Tests/Tests/TestMemStore_testGetNextRow.cs
public class TestMemStore 
{
    [Fact]
    public void TestGetNextRow()
    {
        AddRows(this.memstore);
        Thread.Sleep(1);
        AddRows(this.memstore);
        KeyValue closestToEmpty = this.memstore.GetNextRow(KeyValue.LOWESTKEY);
        Assert.True(KeyValue.COMPARATOR.CompareRows(closestToEmpty, new KeyValue(Bytes.ToBytes(0), DateTime.Now.Ticks)) == 0);
        for (int i = 0; i < ROW_COUNT; i++)
        {
            KeyValue nr = this.memstore.GetNextRow(new KeyValue(Bytes.ToBytes(i), DateTime.Now.Ticks));
            if (i + 1 == ROW_COUNT)
            {
                Assert.Null(nr);
            }
            else
            {
                Assert.True(KeyValue.COMPARATOR.CompareRows(nr, new KeyValue(Bytes.ToBytes(i + 1), DateTime.Now.Ticks)) == 0);
            }
        }
        for (int startRowId = 0; startRowId < ROW_COUNT; startRowId++)
        {
            InternalScanner scanner = new StoreScanner(new Scan(Bytes.ToBytes(startRowId)), FAMILY, int.MaxValue, this.memstore.Comparator, null, memstore.GetScanners());
            List<KeyValue> results = new List<KeyValue>();
            for (int i = 0; scanner.Next(results); i++)
            {
                int rowId = startRowId + i;
                Assert.True(KeyValue.COMPARATOR.CompareRows(results[0], Bytes.ToBytes(rowId)) == 0, "Row name");
                Assert.Equal(QUALIFIER_COUNT, results.Count);
                List<KeyValue> row = new List<KeyValue>();
                foreach (KeyValue kv in results)
                {
                    row.Add(kv);
                }
                IsExpectedRowWithoutTimestamps(rowId, row);
                results.Clear();
            }
        }
    }
}

File: refactored/sleepy-test/java-junit/TestMemStore_testGetNextRow.java
public class TestMemStore extends TestCase {
  public void testGetNextRow() throws Exception {
    addRows(this.memstore);
    // Add more versions to make it a little more interesting.
    // Thread.sleep(1);
    addRows(this.memstore);
    KeyValue closestToEmpty = this.memstore.getNextRow(KeyValue.LOWESTKEY);
    assertTrue(KeyValue.COMPARATOR.compareRows(closestToEmpty,
      new KeyValue(Bytes.toBytes(0), System.currentTimeMillis())) == 0);
    for (int i = 0; i < ROW_COUNT; i++) {
      KeyValue nr = this.memstore.getNextRow(new KeyValue(Bytes.toBytes(i),
        System.currentTimeMillis()));
      if (i + 1 == ROW_COUNT) {
        assertEquals(nr, null);
      } else {
        assertTrue(KeyValue.COMPARATOR.compareRows(nr,
          new KeyValue(Bytes.toBytes(i + 1), System.currentTimeMillis())) == 0);
      }
    }
    //starting from each row, validate results should contain the starting row
    for (int startRowId = 0; startRowId < ROW_COUNT; startRowId++) {
      InternalScanner scanner =
          new StoreScanner(new Scan(Bytes.toBytes(startRowId)), FAMILY,
              Integer.MAX_VALUE, this.memstore.comparator, null,
              memstore.getScanners());
      List<KeyValue> results = new ArrayList<KeyValue>();
      for (int i = 0; scanner.next(results); i++) {
        int rowId = startRowId + i;
        assertTrue("Row name",
          KeyValue.COMPARATOR.compareRows(results.get(0),
          Bytes.toBytes(rowId)) == 0);
        assertEquals("Count of columns", QUALIFIER_COUNT, results.size());
        List<KeyValue> row = new ArrayList<KeyValue>();
        for (KeyValue kv : results) {
          row.add(kv);
        }
        isExpectedRowWithoutTimestamps(rowId, row);
        // Clear out set.  Otherwise row results accumulate.
        results.clear();
      }
    }
  }
}
